### PR TITLE
Android callback

### PR DESCRIPTION
**Pull request by yalab**
Hi.
I added a feature that android call callback.
Thanks.

### CODE REVIEW

File: CCSocialShare/CCSocialManager.h [hunk 1]
```diff
@@ -16,14 +16,19 @@ namespace CCSocialShare {
...
-    
     class SocialManager :public cocos2d::Ref {
     public:
         SocialManager();
         virtual ~SocialManager();
         static bool isAvailable(Service service);
         static void postMessage(Service service, const char* message, SocialManagerCompletionCallback callback);
         static void postMessage(Service service, const char* message, const char* imagePath, SocialManagerCompletionCallback callback);
+#ifdef __ANDROID__
+	static SocialManagerCompletionCallback _callback;
```
Comment (giginet) [inline CCSocialShare/CCSocialManager.h]: You must not use hard tabs.
Please indent by soft tabs.

File: CCSocialShare/CCSocialManager.h [hunk 2]
```diff
@@ -16,14 +16,19 @@ namespace CCSocialShare {
...
+	static void callCallback(int result){
```
Comment (giginet) [inline CCSocialShare/CCSocialManager.h]: You should add spaces before braces.

``` cpp
static void callCallback(PostResult result) {
```

Comment (giginet) [inline CCSocialShare/CCSocialManager.h]: You should use enum type instead of int.

File: CCSocialShare/android/CCSocialManager.cpp [hunk 3]
```diff
@@ -23,10 +23,13 @@ namespace CCSocialShare {
         postMessage(service, message, "", callback);
     }
 
+    SocialManagerCompletionCallback SocialManager::_callback = [](PostResult){};
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: I think you should add space into brace.

``` cpp
SocialManagerCompletionCallback SocialManager::_callback = [](PostResult) { };
```

File: CCSocialShare/android/CCSocialManager.cpp [hunk 4]
```diff
@@ -42,3 +45,11 @@ namespace CCSocialShare {
         }
     }
 };
+
+extern "C"
+{
+	void Java_org_kawaz_socialshare_SocialShare_callCallback(JNIEnv *env, jobject thiz, jint code)
```
Comment (giginet) [inline CCSocialShare/android/CCSocialManager.cpp]: You should use soft tabs(spaces).

File: README.md [hunk 5]
```diff
@@ -141,7 +141,20 @@ LOCAL_C_INCLUDES := $(LOCAL_PATH)/../../../Classes \
                     $(LOCAL_PATH)/../../../Classes/external/CCSocialShare/CCSocialShare
 ```
 
-5: Execute following command
+5: Edit `proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java`
```
Comment (giginet) [inline README.md:144]: Thanks for documentation 👍

File: CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java [hunk 6]
```diff
@@ -89,4 +94,17 @@ static public void copyFile(File src, File dst) throws Exception {
         in.close();
         out.close();
     }
+
+    static public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        if (intent == null) {
+            callCallback(2);
+            return;
+        }
+        // Bundle extras = intent.getExtras();
```
Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: Please remove comment outs.

File: CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java [hunk 7]
```diff
@@ -89,4 +94,17 @@ static public void copyFile(File src, File dst) throws Exception {
         in.close();
         out.close();
     }
+
+    static public void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        if (intent == null) {
+            callCallback(2);
```
Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: I think it is not comfortable to be used hard coded values.
it is better to define `PostResult` type into another file and pass raw value of them.

``` java
public enum PostResult {
    SUCCEED,
    FAILURE,
    CANCELLED
}
```

``` java
callCallback(PostResult::SUCCEED.ordinal());
```

Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: I think it is not comfortable to be used hard coded values.
it is better to define `PostResult` type into another file and pass raw value of them.

```java
public enum PostResult {
    SUCCEED,
    FAILURE,
    CANCELLED
}
```

```java
callCallback(PostResult::SUCCEED.ordinal());
```

File: CCSocialShare/CCSocialManager.h (hunk 2 above)
Comment (giginet) [inline CCSocialShare/CCSocialManager.h]: In my opinion, `callCallback` is not a good name.
I suggest `executeCallback`, `invokeCallback` or etc...

File: CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java [hunk 8]
```diff
@@ -3,6 +3,9 @@
 import android.content.Intent;
 import android.net.Uri;
 import android.os.Environment;
+import android.util.Log;
```
Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: unnecessary

File: CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java [hunk 9]
```diff
@@ -3,6 +3,9 @@
...
+import android.app.Activity;
+import android.os.Bundle;
```
Comment (giginet) [inline CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java]: unnecessary

**Comment by giginet** [comment]
Thank you for your contribution!

I checked your code works correctly.
However, I noted some points. Could you fix them?

**Comment by yalab** [comment]
I fixed for about your advice.

**Comment by giginet** [comment]
Great job ✨ 

Looks good to me.
I'll merge and release newer version.
Thank you for your contribution 😄

**Pull request merged**